Tribler/community/tunnel/conversion.py

```diff
@@ -58,13 +85,13 @@ def _decode_cell(self, placeholder, offset, data):
 
     def _encode_create(self, message):
         payload = message.payload
-        packet = pack("!IHH20s", payload.circuit_id, len(payload.node_public_key),
-                      len(payload.key), payload.node_id) + payload.node_public_key + payload.key
+        packet = pack("!IHH20s?", payload.circuit_id, len(payload.node_public_key),
+                      len(payload.key), payload.node_id, payload.exit_candidates) + payload.node_public_key + payload.key
```

Comment:
You mean by appending the payloads of create and extend with an ip-field? But there is already support for creating circuits to any 'required_exit'. Are there more benefits for adding such an ip-field to the payload?